File: src/SimpleLevelEditor.Formats.EntityConfig/EntityPropertyDescriptor.cs
namespace SimpleLevelEditor.Formats.EntityConfig;

public sealed record EntityPropertyDescriptor
{
	public required string Name { get; init; }

	public required EntityPropertyTypeDescriptor Type { get; init; }

	public required string? Description { get; init; }

	public EntityPropertyDescriptor DeepCopy()
	{
		return this with { Type = Type.DeepCopy() };
	}
}

File: src/SimpleLevelEditor.Formats/Types/EntityConfig/EntityPropertyDescriptor.fs
namespace SimpleLevelEditor.Formats.Types.EntityConfig

type EntityPropertyDescriptor =
    { Name: string
      Type: EntityPropertyTypeDescriptor
      Description: string option }

    member this.DeepCopy() =
        { this with
            Name = this.Name
            Type = this.Type.DeepCopy()
            Description = this.Description }
